pupil_src/shared_modules/calibration_choreography/base_plugin.py

```diff
@@ -163,147 +189,157 @@ def cleanup(self):
     def current_mode(self) -> ChoreographyMode:
         return self.__current_mode
 
-    @property
-    def current_mode_ui_button(self):
-        if self.__current_mode == ChoreographyMode.CALIBRATION:
-            return self.__ui_button_calibration
-        if self.__current_mode == ChoreographyMode.ACCURACY_TEST:
-            return self.__ui_button_accuracy_test
-        raise NotImplementedError(f"Unsupported choreography mode: {self.__current_mode}")
-
     @property
     def is_active(self) -> bool:
         return self.__is_active
 
-    def start(self):
-        self.__is_active = True
-        if self.__current_mode == ChoreographyMode.CALIBRATION:
-            self.__ui_button_accuracy_test_disable()
-        if self.__current_mode == ChoreographyMode.ACCURACY_TEST:
-            self.__ui_button_calibation_disable()
+    @property
+    def selected_choreography_class(self):
+        return self.__class__
+
+    @selected_choreography_class.setter
+    def selected_choreography_class(self, cls):
+        self.__start_plugin(cls)
+
+    @property
+    def status_text(self) -> str:
+        ui_button = self.__mode_button(self.current_mode)
+        return ui_button.status_text or ""
+
+    @status_text.setter
+    def status_text(self, value: T.Any):
+        value = str(value).strip() if value else ""
+        ui_button = self.__mode_button(self.current_mode)
+        ui_button.status_text = value
+
+    @property
+    def pupil_list(self) -> T.List[dict]:
+        return self.__pupil_list
+
+    @property
+    def ref_list(self) -> T.List[dict]:
+        return self.__ref_list
+
+    def on_choreography_started(self, mode: ChoreographyMode):
         self.notify_all(
-            ChoreograthyNotification(
+            ChoreographyNotification(
                 # TODO: Why the subject is always "calibration.started", and not "accuracy_test.started" for accuracy test mode?
-                # mode=self.__current_mode,
+                # mode=mode,
```

Comment:
@papr Could you please clarify why the previous implementation was sending `calibration.started` notifications, even when the user was triggering an accuracy test? Is it just a copy/paste bug or was it intentional? Also, what parts could depend on the old behavior?

Reply:
Can be renamed to `validation.started`. Other instances of `accuracy_test` should be renamed to `validation`.